File: src/slidingwindow/MeanSubArrayofSizeKBruteforce.java
package slidingwindow;

import java.util.Arrays;

public class MeanSubArrayofSizeKBruteforce {

    public static double[] findAverages(int k, int[] arr) {

        double[] result = new double[arr.length - k + 1];

        for (int i = 0; i <= arr.length - k; i++) {
            // find sum of next 'k' elements
            double sum = 0;
            for (int j = i; j < i + k; j++)
                sum += arr[j];
            result[i] = sum / k; //calculating the average
        }

        return result;
    }

    public static void main(String[] args) {
        int[] arr = {1, 3, 2, 6, -1, 4, 1, 8, 2};
        double[] result = MeanSubArrayofSizeKBruteforce.findAverages(5, arr);

        System.out.println("Averages of subarrays of size k: " + Arrays.toString(result));
    }
}

File: python_codes/sliding_window/MeanSubArrayofSizeKBruteforce.py
def find_averages(k, arr):
	result = []
	for i in range(0, len(arr)-k+1):
		# find the smu of the next 'k' elements
		buf = 0.0
		_sum = [buf := buf + arr[j] for j in range(i, i+k)]
		print(_sum)
		# average
		result.append(_sum[-1]/k)
	return result

def main():
	result = find_averages(5, [1,3,2,6,-1,4,1,8,2])
	print("Averages of subarrays of size k: ", str(result))





if __name__ == "__main__":
	main()
